src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2309,7 +2325,7 @@ bool IsVirtualNeededByTypeDueToPreservedScope (MethodDefinition method)
 			return false;
 		}
 
-		bool IsVirtualNeededByInstantiatedTypeDueToPreservedScope (MethodDefinition method)
+		bool IsMethodNeededByTypeDueToPreservedScope (MethodDefinition method)
 		{
 			if (!method.IsVirtual)
```

Comment:
This should be something like `if (!(method.IsVirtual || method.IsStatic))` - it needs to let static methods through as some may have a "base method" in which case we may need to mark them.